tools/blocksconvert/builder/symbols_iterator.go

```diff
@@ -0,0 +1,176 @@
+package builder
+
+import (
+	"encoding/gob"
+	"io"
+	"os"
+
+	"github.com/golang/snappy"
+	"github.com/prometheus/prometheus/tsdb/errors"
+)
+
+type symbolsIterator struct {
+	files []*symbolsFile
+	errs  errors.MultiError
+
+	// To avoid returning duplicates, we remember last returned symbol.
+	last     string
+	returned bool // Used to avoid skipping first "" symbol.
```

Comment:
Instead of `returned` have you considered `last *string` so that we use `nil` value to distinguish it?